xbmc/cores/VideoPlayer/Process/VideoBuffer.cpp

```diff
@@ -421,14 +421,16 @@ void CVideoBufferManager::ReleasePool(IVideoBufferPool *pool)
 {
   CSingleLock lock(m_critSection);
 
-  for (auto it = m_pools.begin(); it != m_pools.end(); ++it)
+  for (auto it = m_pools.begin(); it != m_pools.end();)
   {
     if ((*it).get() == pool)
     {
       m_discardedPools.push_back(*it);
-      m_pools.erase(it);
+      it = m_pools.erase(it);
```

Comment:
I would just break the loop after the pool was found. No need to search for something already gone.

Reply:
ofc, wasn't sure that it's a dup free list.